include/cppoptlib/solver/lbfgsbsolver.h

```diff
@@ -236,6 +245,7 @@ class LbfgsbSolver : public ISolver<TProblem, 1> {
       const Scalar rate = MoreThuente<TProblem, 1>::linesearch(x,  SubspaceMin-x ,  problem, alpha_init);
       // update current guess and function information
       x = x - rate*(x-SubspaceMin);
+      clamp_x(problem, x);
```

Comment:
Well, L-BFGS-B should be the solver with inherently can deal with constraints (cauchy-point). But as I said above, if this helps on the benchmark. I am fine with that. I just need to run the benchmark later on.